File: kernel/drivers/agnostic/null.lua
local BaseChrDev = require("drivers.common.basechrdev")

---@class Null: BaseChrDev
local Null = BaseChrDev:new()

function Null:read_file(inode, offset, length)
    return ""
end

function Null:write_file(inode, offset, data)
    return #data
end

return Null

File: kernel/drivers/agnostic/null.ts
import { BaseChrDev } from "../common/basechrdev";
import { Inode } from "../../vfs/inode/init";

export class Null extends BaseChrDev {
	read_file(inode: Inode, offset: integer, length: integer): string {
		return "";
	}

	write_file(inode: Inode, offset: integer, data: string): integer {
		return data.length;
	}

	ioctl(inode: Inode, request: integer, arg: any): unknown {
		return;
	}
}
